src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -339,12 +340,24 @@ public String getShortDescription() {
 
     @Exported
     @Override public boolean isConcurrentBuild() {
-        return !Boolean.FALSE.equals(concurrentBuild);
+        WorkflowConcurrentBuildJobProperty p = getProperty(WorkflowConcurrentBuildJobProperty.class);
+        if (p != null) {
+            return p.getAllowConcurrentBuilds();
+        } else {
+            /* settings compatibility */
+            return !Boolean.FALSE.equals(concurrentBuild);
+        }
     }
     
     public void setConcurrentBuild(boolean b) throws IOException {
-        concurrentBuild = b ? null : false;
```

Comment:
I think you now need to unconditionally set `concurrentBuild` to null. And deprecate this method.

Either way, you should start off by writing the `@LocalData`-based tests which define what you expect to happen w.r.t. settings compatibility.